### ISSUE TITLE
Make functions accept StateVector in addition to Numpy array

### ISSUE THREAD

**Issue by kevinsung** [qiskit-community/ffsim#253]
Now that the [StateVector](https://github.com/qiskit-community/ffsim/blob/156f1ecb097f770953c7c5b80c16dcaecc4bdd31/python/ffsim/states/states.py#L30) class exists, functions that accept `vec` as a Numpy array should be updated to also accept a StateVector, in which `norb` and `nelec` don't also need to be passed.

Non-exhaustive list of affected modules:
- [ ] `gates/`
- [ ] `protocols/`

**Comment by tsmanral** [comment]
Hi @kevinsung. Does this issue still exists since this is a very old Issue? If yes, I'd like to assign it to myself and contribute. Please let me know.

**Comment by kevinsung** [comment]
> Hi @kevinsung. Does this issue still exists since this is a very old Issue? If yes, I'd like to assign it to myself and contribute. Please let me know.

@tsmanral Yes, this issue still exists. For an example in the `gates` module, `apply_orbital_rotation` still only accepts a Numpy array, and should be update to support StateVector:

https://github.com/qiskit-community/ffsim/blob/72fe5c3c457c2ba964f6ec7fd49896167347808b/python/ffsim/gates/orbital_rotation.py#L45

Note that StateVector now has an `__array__` method, so many Numpy functions should work on it directly.

There's no need for me to assign the issue to you. Feel free to go ahead and open a pull request!

**Comment by tsmanral** [comment]
Thank you @kevinsung. I've forked this repo: https://github.com/tsmanral/ffsim/tree/fix-accpt-StateVector and will create a PR to the main after I update it with StateVectors.
